src/ripple/app/ledger/LedgerMaster.cpp

```diff
@@ -959,7 +947,7 @@ class LedgerMasterImp
                     WriteLog (lsTRACE, LedgerMaster) << "tryAdvance discovered missing " << missing;
                     if ((missing != RangeSet::absent) && (missing > 0) &&
                         shouldAcquire (mValidLedgerSeq, ledger_history_,
-                            ledger_history_index_, missing) &&
+                            getApp ().getSHAMapStore ().getCanDelete (), missing) &&
```

Comment:
Be aware that we are creating cyclic dependencies in call graphs. LedgerMaster depends on SHAMapStore, and SHAMapStore depends on LedgerMaster. This tight coupling is a code smell.